pkg/regexp/regexp.go

```diff
@@ -0,0 +1,210 @@
+package regexp
+
+import (
+	"io"
+	"regexp"
+	"sync"
+)
+
+// Regexp is a wrapper struct used for wrapping MustCompile regex expressions
+// used init. Using this stucture should help speed up startup time of apps
+// that want to initialize their regex at init time.
+type Regexp struct {
+	once   sync.Once
+	regexp *regexp.Regexp
+	val    string
+}
+
+func New(val string) Regexp {
+	re := Regexp{
+		val: val,
+	}
+	if precompile {
+		re.compile()
+	}
+	return re
+}
+
+func (re *Regexp) compile() {
+	re.once.Do(func() {
```

Comment:
Don’t do this if `precompile`?

On the one hand, the current version makes run-time profiling of the `precompile` version representative of the other one.

On the other hand, making compilation conditional here would allow an easy experiment to see if precompilation, _in total_, pays off in a particular situation.

Reply:
Ok, so you want the re.once.Do command not to be executed.